### PR TITLE
Allow containers to reach the host.

### PR DESCRIPTION
**Pull request by spikecurtis**
Starting in Felix v0.25, the default behaviour is to prevent
endpoints from reaching thier local host IP.  However, this is
required for common containerized environments like Mesos,
where the containerized executor exchanges messages with a
Mesos Slave process on the host.

This change reverts Felix to the pre-0.25 behaviour by setting
a configuration field in etcd.

It also includes an ST for regression testing.

### CODE REVIEW

File: calico_containers/tests/st/test_container_to_host.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            ep1 = host.calicoctl("container %s endpoint-id show" % node1)
+
+            # Now add the profiles.
+            host.calicoctl("endpoint %s profile set TEST" % ep1)
+
+            # Check it works.  Note that the profile allows all outgoing
+            # traffic by default, and conntrack should allow the reply.
+            node1.assert_can_ping(host.ip, retries=10)
+
+            # Test the teardown commands
```
Comment (tomdee) [inline calico_containers/tests/st/test_container_to_host.py]: No need to test these here

File: calico_containers/tests/st/test_container_to_host.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+from test_base import TestBase
+from tests.st.utils.docker_host import DockerHost
+
+
+class TestContainerToHost(TestBase):
+    def test_container_to_host(self):
+        """
+        Test that a container can ping the host.  (Without using the docker
+        network driver, since it doesn't support that yet.)
```
Comment (tomdee) [inline calico_containers/tests/st/test_container_to_host.py]: I don't understand this statement. How doesn't the network driver support this?

**Comment by tomdee** [comment]
A comment on the tests. Otherwise LGTM

**Comment by frostynova** [comment]
Can this patch back port to 0.4.8? We do not want to try libnetwork now.

**Comment by spikecurtis** [comment]
@flex are you working with Mesos, or some other orchestration system that needs communication from container to host?

**Comment by frostynova** [comment]
@spikecurtis  Sorry, it seems v0.4.8 is not affected by this, so i delete my comment.

Our team is building a PaaS, some component need to communicate with etcd running on the host.

**Pull request merged**